### PR TITLE
Problem: (CRO-294) Client can generate invalid transactions

### PR DESCRIPTION
**Pull request by devashishdxt**
Solution: Added balance checks when creating network ops transactions

### CODE REVIEW

**Comment by devashishdxt** [comment]
> looks ok, but is it only for staked state operations?
> e.g. if one sends TX that spends some tx inputs... if the client doesn't sync, those inputs would still be "unspent" in the internal state (rather than "potentially spent" with some TXID to check whether it was really committed), is that correct?

That is correct. Client won't know if some inputs are spent until it syncs the block in which they were spent.

**Comment by tomtau** [comment]
> > looks ok, but is it only for staked state operations?
> > e.g. if one sends TX that spends some tx inputs... if the client doesn't sync, those inputs would still be "unspent" in the internal state (rather than "potentially spent" with some TXID to check whether it was really committed), is that correct?
> 
> That is correct. Client won't know if some inputs are spent until it syncs the block in which they were spent.

ok, I created CRO-524 for that

File: client-core/src/wallet/default_wallet_client.rs [hunk 1]
```diff
@@ -395,6 +395,19 @@ where
...
     }
 
+    fn has_unspent_transactions(
+        &self,
+        name: &str,
+        passphrase: &SecUtf8,
+        inputs: &[TxoPointer],
+    ) -> Result<bool> {
+        // Check if wallet exists
+        self.wallet_service.view_key(name, passphrase)?;
```
Comment (calvinaco) [inline client-core/src/wallet/default_wallet_client.rs]: A suggestion: I notice we are using `view_key` to check for wallet existence or passphrase correctness in wallet client, maybe we add a `wallet_service.has_wallet()` ?

**Comment by devashishdxt** [comment]
bors retry

**Comment by tomtau** [comment]
bors retry

**Pull request merged**